libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -328,6 +328,7 @@ size_t UARTDriver::write(const uint8_t *buffer, size_t size)
         /*
           use the per-byte delay loop in write() above for blocking writes
          */
+        _write_mutex.give();
```

Comment:
this doesn't look right, we end up calling give() twice

Reply:
Fixed.